adapters/grid/grid.go

```diff
@@ -37,6 +54,21 @@ func processImp(imp *openrtb2.Imp) error {
 	return nil
 }
 
+func setImpExtData(imp openrtb2.Imp) openrtb2.Imp {
+	var ext ExtImp
+	if err := json.Unmarshal(imp.Ext, &ext); err != nil {
+		return imp
+	}
+	if ext.Data != nil && ext.Data.AdServer != nil && ext.Data.AdServer.AdSlot != "" {
+		ext.Gpid = ext.Data.AdServer.AdSlot
+	}
+	extJSON, err := json.Marshal(ext)
```

Comment:
You should put this marshalling up in the above if clause. If you have made no changes to the ext data, there is no reason to remarshal it that I can see.

Reply:
Done